exp/controllers/machinepool_controller_phases.go

```diff
@@ -108,6 +108,33 @@ func (r *MachinePoolReconciler) reconcileExternal(ctx context.Context, cluster *
 		return external.ReconcileOutput{Paused: true}, nil
 	}
 
+	if strings.HasSuffix(ref.Kind, external.TemplateSuffix) {
+		owner := &metav1.OwnerReference{
+			APIVersion: expv1.GroupVersion.String(),
+			Kind:       "MachinePool",
+			Name:       m.Name,
+			UID:        m.UID,
+		}
+		ref, err = external.CloneTemplate(ctx, &external.CloneTemplateInput{
+			Client:      r.Client,
+			TemplateRef: ref,
+			Namespace:   m.Namespace,
+			ClusterName: cluster.Name,
+			OwnerRef:    owner,
+		})
+		if err != nil {
+			return external.ReconcileOutput{}, errors.Wrap(err, "failed to clone template")
+		}
+		obj, err = external.Get(ctx, r.Client, ref, m.Namespace)
+		if err != nil {
+			if apierrors.IsNotFound(errors.Cause(err)) {
+				return external.ReconcileOutput{}, errors.Wrapf(err, "could not find %v %q for MachinePool %q in namespace %q, requeuing",
```

Comment:
Should this be could not find the MachinePool template?
Also, for my understanding, what kind of error are we trying to catch by getting the cloned template after it is cloned? If the clone operation passed, the template should be there...

Reply:
@fabriziopandini I may be wrong because I don't have a deep knowledge of the codebase of the cluster-api, but by my understanding the object to be reconciled must be the concrete object and not the template so I do Get after the clone

Reply:
obj variable is returned to outside no? https://github.com/kubernetes-sigs/cluster-api/blob/1f87ca9cc54e5c22217bf09234a604450b1b91b1/exp/controllers/machinepool_controller_phases.go#L192